### DISCUSSION TITLE
Feather Standard Pinout Extension & Spec Interpretation

### DISCUSSION THREAD

**Discussion by 0Grit** [esp-rs/esp-rust-board#7] [Ideas]
See #3 for initial context of this discussion

The Feather pin spec leaves a lot to be desired for professional use cases. (As expected) But it seems to be used often so lets see if we can improve upon it in a backwards compatible way.

Feather Spec is here: https://learn.adafruit.com/adafruit-feather/feather-specification
The spec does not do much to constrain potential interpretations. I will leave comments for my own interpretations in order to enable them to be discussed individually.

Below I have created a table that shows the pinout and have summarily described the pin functions. I'd like the outcome of the discussion to be

* How to involve Adafruit if desired
* Which pins to add
* Where to add them
* Ensure the additions will be backward compatible

| Left Header | Function                                                     | ⬛ | Right Header | Function                                                                                                      |
|:-----------:|--------------------------------------------------------------|---|:------------:|---------------------------------------------------------------------------------------------------------------|
|     Rst     | Active low; "main feather board reset line"                  | ⬛ |    No Pin    |                                                                                                               |
|     3.3V    | 3.3V pin powers wings. Take VUSB/VBAT to 3.3V                | ⬛ |    No Pin    |                                                                                                               |
|     Aref    | Analog reference, usually for the host processor             | ⬛ |    No Pin    |                                                                                                               |
|     GND     | This is the COMMON reference point                           | ⬛ |    No Pin    |                                                                                                               |
|      A0     | Analog 0                                                     | ⬛ |      Bat     | LiPoly battery voltage (3.0-3.7V) (Battery being present should result in VUSB being disconnected             |
|      A1     | Analog 1                                                     | ⬛ |      En      | Input to mainboard; must depower the 3.3V output. PLEASE CONNECT THIS TO A GPIO OF THE HOST PROCESSOR AS WELL |
|      A2     | Analog 2                                                     | ⬛ |      USB     | ?!?!? Spec refers to three different nets that could actually be the same thing (USB, VBUS, VUSB)             |
|      A3     | Analog 3                                                     | ⬛ |      D13     | Digital IO (User LED)                                                                                         |
|    A4/D24   | Analog 4 (D24 DIGITAL IO ONLY IF NO ANALOG AVAILABLE)        | ⬛ |      D12     | Digital IO                                                                                                    |
|    A5/D25   | Analog 5 (D25 DIGITAL IO ONLY IF NO ANALOG AVAILABLE)        | ⬛ |      D11     | Digital IO                                                                                                    |
|     SCK     | Serial CLock (SPI)                                           | ⬛ |      D10     | Digital IO                                                                                                    |
|      MO     | Manager out, Subordinate In (SPI)                            | ⬛ |      D9      | Digital IO                                                                                                    |
|      MI     | Manager In, Subordinate out (SPI)                            | ⬛ |      D6      | Digital IO                                                                                                    |
|    RX/D0    | I will assume this is application RX (Not Debug RX)          | ⬛ |      D5      | Digital IO                                                                                                    |
|    TX/D1    | I will assume this is application TX (Not Debug TX)          | ⬛ |      SCL     | Serial Clock Line (I2C) (They recommend having the wings pull on the i2c lines)                               |
|    :star:   | "Free Pin", more like "zombie pin" This may as well be DNP.  | ⬛ |      SDA     | Serial Data line (I2C) (They recommend having the wings pull in the i2c lines)                                |

**Comment by 0Grit** [comment]
@pedrominatel when I originally was drafting this discussion I had a good half hour's worth of comments that github lost when I realized I hadn't selected a category yet and selected one. (Now a known discussions bug)

Is it too late for me to capture those notes/comments here?

**Reply by MabezDev** [reply]
@0Grit Sorry to hear your comments got lost, we would definitely appreciate your input! It's definitely not too late :)

If you'd like to talk about the board myself and pedro will be in the [community meeting](https://github.com/esp-rs/rust/discussions/94) which you are welcome to join.

**Comment by pedrominatel** [comment]
Hi @0Grit, I've used this proposal as a starting point for the proposal at [Pin Layout](https://github.com/esp-rs/esp-rust-board#pin-layout).

Feel free to let your comments there, or prepare a PR.

Thanks!
